# 小组2-实验1-中期检查-个人实验记录

> 实验一: 全基因组基因的从头预测及结构建模

> 1730416009 朱泽峰

## 分工任务

5. 利用`gffcompare`比较不同软件的预测结果，进行挑选与整合
   * 解读`gffcompare`结果
   * 两两共同特征的选择
   * 而后进行挑选与整合
6. 将4和5的结果与1下载的原始注释文档进行比对，分析异同，评估优劣，改良

## 实验记录

### 不同软件预测结果文件的处理安排

在其他小组成员分别利用`Augustus,Geneid,GeneMark`对`Paramecium tetraurelia,Caenorhabditis elegans,Saccharomyces cerevisiae`三个物种的基因组数据进行分析后，得到了软件输出的GFF与GTF格式文件。预先设定的分析流程是按照`Augustus`的GFF输出文件内容来安排，即提取其中的蛋白序列信息于建立好的`blast`数据库进行搜索，找到对应的`UniProt`条目进而重注释例如`Augustus`输出的结果，方便后续进行对比分析。

但是在实际实验过程中发现，`Geneid,GeneMark`最新版本输出结果中(GFF3格式或GTF格式文件)没有所需的蛋白序列信息。因此安排负责同学对软件参数与版本进行调查，最终决定回退至特定版本，以获得序列信息。

### 各软件GFF文件经过重注释的结果与原GFF文件进行对比

#### 安装最新版软件

```bash
git clone https://github.com/gpertea/gffcompare
cd gffcompare
make release
```

#### 软件运行

接着分别设定各个物种的参考GFF文件以及各个软件的对应GFF结果文件，运行`gffcompare`

命令运行示例如下:


```bash
zzf@LAPTOP-MGPQT3HG:~/software/gffcompare$ ./gffcompare -V -r $gff_sc_r $gff_sc_au -o /mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc
Prefix for output files: augustus_compare_sc
Loading reference transcripts..
  6445 reference transcripts loaded.
  1 duplicate reference transcripts discarded.
Warning: adjusted transcript g1570.t1 boundaries according to terminal exons.
Warning: adjusted transcript g2180.t1 boundaries according to terminal exons.
  5465 query transfrags loaded.
Cleaning up..
Done.
```

输出结果示例如下:

```bash
zzf@LAPTOP-MGPQT3HG:~/software/gffcompare$ ls /mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc*
/mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc.annotated.gtf  /mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc.stats
/mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc.loci           /mnt/c/GitWorks/BIOI-ST/Genomics/augustus_compare_sc.tracking
```

#### `Augustus`结果分析

<table>
    <tr><td>Saccharomyces cerevisiae</td><td>Caenorhabditis elegans</td><td>Paramecium tetraurelia</td></tr>
    <tr>
        <td>

     Query mRNAs :    5465 in    5465 loci  (382 multi-exon transcripts)
            (0 multi-transcript loci, ~1.0 transcripts per locus)
     Reference mRNAs :    6444 in    6342 loci  (341 multi-exon)
     Super-loci w/ reference transcripts:     5440

                       | Sensitivity | Precision  |
             Base level:    96.9     |    99.4    |
             Exon level:    78.3     |    90.5    |
           Intron level:    50.4     |    42.2    |
     Intron chain level:    49.3     |    44.0    |
       Transcript level:    80.4     |    94.8    |
            Locus level:    81.7     |    94.8    |

     Matching intron chains:     168
       Matching transcripts:    5183
              Matching loci:    5181

          Missed exons:    1009/6799	( 14.8%)
           Novel exons:      83/5889	(  1.4%)
        Missed introns:      116/355	( 32.7%)
         Novel introns:      183/424	( 43.2%)
           Missed loci:     830/6342	( 13.1%)
            Novel loci:      11/5465	(  0.2%)

     Total union super-loci across all input datasets: 5451

        
</td>
<td>

     Query mRNAs :    6445 in    6445 loci  (6058 multi-exon transcripts)
                (0 multi-transcript loci, ~1.0 transcripts per locus)
     Reference mRNAs :   56876 in   41076 loci  (29689 multi-exon)
     Super-loci w/ reference transcripts:     5077
                      | Sensitivity | Precision  |
            Base level:    23.3     |    85.7    |
            Exon level:    16.5     |    77.2    |
          Intron level:    22.2     |    87.2    |
    Intron chain level:     7.9     |    38.7    |
      Transcript level:     4.2     |    37.0    |
           Locus level:     5.7     |    37.0    |

     Matching intron chains:    2342
       Matching transcripts:    2387
              Matching loci:    2358

          Missed exons:  139180/177965	( 78.2%)
           Novel exons:    2364/36738	(  6.4%)
        Missed introns:   88354/118707	( 74.4%)
         Novel introns:    1585/30293	(  5.2%)
           Missed loci:   35685/41076	( 86.9%)
            Novel loci:     406/6445	(  6.3%)

     Total union super-loci across all input datasets: 5483

        
</td>
<td>

     Query mRNAs :   12983 in   12983 loci  (10108 multi-exon transcripts)
            (0 multi-transcript loci, ~1.0 transcripts per locus)
     Reference mRNAs :   49143 in   48890 loci  (31615 multi-exon)
     Super-loci w/ reference transcripts:    12022
                       | Sensitivity | Precision  |
             Base level:    26.9     |    95.6    |
             Exon level:     3.4     |    12.1    |
           Intron level:     0.0     |     0.0    |
     Intron chain level:     0.0     |     0.0    |
       Transcript level:     1.1     |     4.3    |
            Locus level:     1.1     |     4.3    |

     Matching intron chains:       0
       Matching transcripts:     560
              Matching loci:     560

          Missed exons:  100350/139401	( 72.0%)
           Novel exons:    3051/39726	(  7.7%)
        Missed introns:   63707/90258	( 70.6%)
         Novel introns:    8565/26743	( 32.0%)
           Missed loci:   31430/48890	( 64.3%)
            Novel loci:     364/12983	(  2.8%)

     Total union super-loci across all input datasets: 12386 

        
</td>
    </tr></table>

可以看到，

* 对于`Saccharomyces cerevisiae`，除去内含子层面的预测，`Augustus`在Precision层面(即给出的预测结果的正确率)以及Sensitivity层面(即正确预测出的结果于参考GFF中的占比)的预测都足够理想
* 对于`Caenorhabditis elegans`，在Sensitivity层面的预测不够良好，即正确预测出的结果于参考GFF中的占比较低，且转录本层面的Precision也不够理想
* 对于`Paramecium tetraurelia`，仅BaseLevel的Precision数值较高，其余数值都很低，说明`Augustus`选用的模型对于本次实验的数据材料存在比较明显的错配